scripts/run_frontend_tests.sh

```diff
@@ -51,6 +51,12 @@ export DEFAULT_SKIP_INSTALLING_THIRD_PARTY_LIBS=false
 export DEFAULT_RUN_MINIFIED_TESTS=false
 maybeInstallDependencies "$@"
 
+# The following command finds all the spec files (except known failing files)
+# and merges them all to a single file which Karma uses to run its tests. The
+# Karma is unable to run the tests on multiple files and the DI fails in that
+# case, the reason of which is unclear.
+find . -type f \( -name "*.spec.ts" ! -name "state-interaction-editor.directive.spec.ts" -o -name "*Spec.ts" \) -exec cat {} \; > ./core/templates/dev/head/combined-tests.spec.ts
```

Comment:
Why is there an exception for the `state-interaction-editor.directive.spec.ts"` file?

Reply:
I was failing the Karma. Also, this is a file exempted from Karma testing since webpack came in.

Reply:
The other ones exempted in `karma.conf.ts` passed?

Reply:
Yes, they did.

Reply:
Okay, explain that in the comment too.

Reply:
Okay scratch that I have fixed that too. Now there are no exceptions. I have removed the files from the karma conf too :)

Reply:
Great, good job!